### PR TITLE
Update to SDK 23 (Android 6)

### LINKED ISSUE #515: Update build dependencies
* Update to the current build tools and sdk
* Update dependencies like the support library
* Check if the -SNAPSHORT versions are still needed
* Clean up build.gradle

### PR DESCRIPTION
**Pull request by audax**
Resolves #515

### CODE REVIEW

**Comment by enkore** [comment]
Looks like the GHPRB trigger is broken here as well (-accounting has the same problem atm)

**Comment by audax** [comment]
Testcoment

**Comment by Rehwanne** [comment]
retest this please

File: qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/files/OpenImageUITest.java [hunk 1]
```diff
@@ -83,13 +84,13 @@ public void cleanUp() {
     }
 
     @Test
-    public void testOpenFiles() {
-        testFile("file1.jpg");
-        pressBack();
+    public void testOpenFiles() throws Throwable {
+
+        mPicassoIdlingResource.init(mActivity);
```
Comment (JoGir) [inline qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/files/OpenImageUITest.java]: The activitiy is already attached to the idlingResource in the setUp method

**Comment by audax** [comment]
Apparently either the selectors in RecyclerViews or the longClick action do not work on real devices with SDK 23. The freshly ignored tests run just on an emulator.

**Comment by JoGir** [comment]
I think its the recyclerView. I changed the actions to
```
onView(withId(R.id.contact_list)).check(matches(isDisplayed()));
onView(withText(CONTACT_1)).perform(longClick());
```
and it runs more stable

File: qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/ImportExportContactsUITest.java [hunk 2]
```diff
@@ -67,7 +61,9 @@
     public void setUp() throws Exception {
         super.setUp();
         createTestContacts();
-        launchActivity(null);
+        Intent intent = new Intent(Intent.ACTION_MAIN);
+        intent.putExtra(MainActivity.START_CONTACTS_FRAGMENT, true);
+        mActivity = mActivityTestRule.launchActivity(intent);
```
Comment (JoGir) [inline qabelbox/src/androidTest/java/de/qabel/qabelbox/ui/ImportExportContactsUITest.java]: pls call launchActivity on the AbstractUITest Class not on the testRule

**Comment by audax** [comment]
`onView(withText(CONTACT_1)).perform(longClick());`

I tried that one, it works on the emulator but not the real device, still.

**Comment by audax** [comment]
Well maybe it does work after all.

**Pull request merged**